DuckDuckGo/HomeRowReminder.swift

```diff
@@ -54,13 +60,21 @@ class HomeRowReminder {
         return storage.shown
     }
 
-    private func hasReminderTimeElapsed() -> Bool {
-        guard let date = storage.firstAccessDate else {
-            storage.firstAccessDate = Date()
-            return false
+    private func hasReminderTimeElapsed(isDefaultBrowserSupported: Bool) -> Bool {
+        if isDefaultBrowserSupported {
+            guard let date = homeMessageStorage.dateDismissed(forHomeMessage: .defaultBrowserPrompt) else {
+                return false
+            }
+            let days = abs(date.timeIntervalSinceNow / 24 / 60 / 60)
+            return days > Constants.reminderTimeInDaysIOS14
```

Comment:
Could this be encapsulated in this storage class behind a "hasExpired" function?  In similar situations I've used `shouldShow` and that would usually encapsulate both dismissed and expiry logic.